### ISSUE TITLE
Hibari v0.1.11 Release

### ISSUE THREAD

**Issue by tatsuya6502** [hibari/hibari#49]
Current stable release Hibari v0.1.10 doesn't work on Erlang/OTP 17. So prepare Hibari v0.1.11 that is compatible with OTP 17.

v0.1.11 will also include many features originally scheduled for v0.3.0. I may remove some of them, but the followings are the current changes on the `dev` branch.

**NEW FEATURES**
- hibari >> GH18 - Add DTrace tracepoints for Erlang/OTP R15 or later
- hibari >> GH19 - Introduce Basho Lager as the primary logging facility
- gdss-brick >> GH2 - brick_server new client API - `rename`
- gdss-brick >> GH13 - Add `attrib` and `exp_time` directive flags to `rename/6`
- gdss-brick >> GH14 - Add `attrib` and `exp_time` directive flags to `replace/6` and `set/6`

**FIXES**
- hibari >> GH47 - reltool command will fail to generate package due to broken symlinks created by repo command
- gdss-client >> GH8 - `brick_simple_client:find_the_brick/3` crashes for unknown table

**ENHANCEMENTS**
- gdss-client >> GH2 - `brick_simple:{add,set,replace}` APIs do not return the server-side timstamp for success

**OTHERS**
- hibari >> GH20 - Remove riak-err
- hibari >> GH21 - Add gmt_elog module again
- hibari >> GH23 - Introduce qc wrapper
- hibari >> GH24 - Update for Erlang/OTP R16
- hibari >> GH27 - Support for building and running on Joyent SmartOS
- hibari >> GH37 - Update for Erlang/OTP 17.x (Remove ubf and ubf-thrift)
- hibari >> GH39 - Drop support for older Erlang/OTP releases

**Comment by tatsuya6502** [comment]
In order to make server side rename to work correctly during scavenger run, gdss-brick project's `brick_ets` module needs `is_sequence_frozen/1` and `is_sequence_frozen/2` functions from this commit https://github.com/hibari/gdss-brick/commit/7bb150abaec0a86556d2d42c32d179ead7c2f28e.

They are in the huge commit (4 changed files  with 637 additions and 686 deletions) with many refactoring stuff so it can't be cherry-picked.

**Comment by tatsuya6502** [comment]
> In order to make server side rename to work correctly during scavenger run, gdss-brick project's `brick_ets` module needs `is_sequence_frozen/1` and `is_sequence_frozen/2` functions from this commit

I addressed this last night ([link](https://github.com/hibari/gdss-brick/issues/2#issuecomment-78395202)) and ran basho_bench with `rename/6` operation for 6 hours. It included a Scavenger run at 3 am and basho_bench reported no error. It looks good.

**Comment by tatsuya6502** [comment]
> v0.1.11 will also include many features originally scheduled for v0.3.0. I may remove some of them, but the followings are the current changes on the dev branch.

I merged dev branch into release-0.1.11 branch. I'll update the release notes (move many items from v0.3.0 to v0.1.11).

**Comment by tatsuya6502** [comment]
I'm currently investigating the following QuickCheck test failure, which was found before v0.1.10 release.

- [gdss-admin >> GH4 - QuickCheck squorum_eqc_tests:prop_squorum1 fails with quorum_error](https://github.com/hibari/gdss-admin/issues/4)

**Comment by tatsuya6502** [comment]
> I'm currently investigating the following QuickCheck test failure, which was found before v0.1.10 release.
> 
> - [gdss-admin >> GH4 - QuickCheck squorum_eqc_tests:prop_squorum1 fails with quorum_error](https://github.com/hibari/gdss-admin/issues/4)

I first thought this would be a false alert, but it actually is a real bug. I opened the following issue.

- [gdss-admin >> GH8 - Can't bootstrap Hibari cluster in multi admin node config](https://github.com/hibari/gdss-admin/issues/8)

**Comment by tatsuya6502** [comment]
>> I'm currently investigating the following QuickCheck test failure, which was found before v0.1.10 release.
>> 
>> - [gdss-admin >> GH4 - QuickCheck squorum_eqc_tests:prop_squorum1 fails with quorum_error](https://github.com/hibari/gdss-admin/issues/4)
>
> I first thought this would be a false alert, but it actually is a real bug. I opened the following issue.
>
> - [gdss-admin >> GH8 - Can't bootstrap Hibari cluster in multi admin node config](https://github.com/hibari/gdss-admin/issues/8)

I was both right and wrong on the point.

- One reason that the QuickCheck test failure is an invalid test scenario, which brings down and up majority of squorum bricks without running repair. This causes data loss.
- Another reason was v0.1.11's new feature "server side timestamp". `brick_squorum` needs to be update to avoid to use server-side timestamp. This will be done by a part of hibari/gdss-client#2.

**Comment by tatsuya6502** [comment]
OK. Now the following issue is resolved.

- [gdss-admin >> GH8 - Can't bootstrap Hibari cluster in multi admin node config](https://github.com/hibari/gdss-admin/issues/8)

I'll spend the rest of this week for pre-release testing, and hopefully, I can release v0.1.11 in this weekend.

**Comment by tatsuya6502** [comment]
- [hibari >> GH38 - Upgrade UBF to version 2.0 and re-enable gdss-ubf-proto](https://github.com/hibari/hibari/issues/38)

I tried if I can get this done quickly for v0.1.11. It seems possible. At least I can run UBF client against Hibari now. (I don't have Thrift installed on my FreeBSD machine, so I can't generate Hibari Thrift client right now to test the Thrift protocol server.)

I'll start to push my local changes soon.

```erl
Erlang/OTP 17 [erts-6.3] [source] [64-bit] [smp:8:8] [async-threads:64] [hipe] [kernel-poll:true]

Eshell V6.3  (abort with ^G)
(hibariclient@127.0.0.1)1> {ok, P1, _} = ubf_client:connect("localhost", 7581, [{proto, ubf}], 5000).
{ok,<0.43.0>,{'#S',"gdss_meta_server"}}
(hibariclient@127.0.0.1)2> ubf_client:rpc(P1, {startSession, {'#S', "gdss"}, []}).
{reply,{ok,ok},none}
(hibariclient@127.0.0.1)3> ubf_client:rpc(P1, {set, tab1, <<"foo">>, <<"foo val">>, 0, [], 5000}).
{reply,{ok,1427383568427451},none}
(hibariclient@127.0.0.1)4> ubf_client:rpc(P1, {get, tab1, <<"foo">>, [], 5000}).
{reply,{ok,1427383568427451,<<"foo val">>},none}
(hibariclient@127.0.0.1)5> ubf_client:rpc(P1, {bbb, 3000}).
{reply,{clientBrokeContract,{bbb,3000},[]},none}
(hibariclient@127.0.0.1)6> ubf_client:stop(P1).
ok
(hibariclient@127.0.0.1)7>
```

**Comment by tatsuya6502** [comment]
I pushed local changes for ubf.

I ran basho_bench (with Erlang native API client, not ubf) for 12 hours against a 4-node Hibari cluster (with chain length 3 and 3 admin nodes). It looks good. Also I merged current stuff on dev branch into release-0.1.11 branch.

I want to clean up docs on http://hibari.github.com/hibari-doc/ before release. I need one more week?

**Comment by tatsuya6502** [comment]
Updated the release notes: https://github.com/hibari/hibari/blob/release-0.1.11/relnotes/v0.1.11.txt

**Comment by tatsuya6502** [comment]
Updated the hibari-doc website: http://hibari.github.io/hibari-doc/

Notes:
- index.*.html still need some updates
- System Administrator's Guide is not building
- hibari project's README also needs some updates

**Comment by tatsuya6502** [comment]
Updated the README of hibari project:
https://github.com/hibari/hibari/blob/dev/README.md

**Comment by tatsuya6502** [comment]
- Updated the index.en.html
  * http://hibari.github.io/hibari-doc/
- Updated the change log
  * http://hibari.github.io/hibari-doc/webpage-CHANGELOG.html

System Administrator's Guide is still not building, and I decided to leave it because it will be soon(?) replaced by [this one](https://hibari.readthedocs.org/en/latest/admin-guide/index.html) on Read The Docs. ([Link to the GitHub issue](https://github.com/hibari/hibari/issues/50))

**Comment by tatsuya6502** [comment]
Everything look good and released 0.1.11.
